unittests/data/type_traits_castxml.hpp

```diff
@@ -195,8 +195,8 @@ namespace no{
     typedef std::set< std::string > string_set_type;
     typedef std::multimap< std::string, std::string > s2s_multimap_type;
     typedef detail::vertex vertex_type;
-    typedef detail::const_item const_item_t;
-    typedef detail::const_container const_container_t;
+    // typedef detail::const_item const_item_t;
```

Comment:
These tests now fail, and I'm not sure whether they were correct to begin with. `detail::const_item` contains a single data member - `const int values[10]`. So `values` is copy constructible, but it is not copy assignable. 

The question is what does `is_noncopyable` indicate? Is it supposed to be *not copy-constructible OR not copy-assignable*? In that case these two tests should be moved to the `yes` section.

Or is it supposed to be *not copy-constructible AND not copy-assignable*? If this is the case, then the logic for detecting noncopyable types needs to be modified. It currently iterates through data members and if any are `const`, says the type is noncopyable. But it should really be checking for both not copy constructible and not copy assignable for all data members.